R/cube-dims.R

```diff
@@ -39,6 +39,25 @@ cubeVarReferences <- function (x) {
     if (tuple$type == "enum" && "subreferences" %in% names(tuple)) {
         tuple$type <- "subvariable_items"
     }
+
+    if (!is.null(tuple$subreferences)) {
+        # inject subreference names into the tuple if they exist. Default to ""
+        # if null (for backwards compatibilitiy with cubes that have __any__
+        # __all__ and __none__)
+        tuple$subvariables <- vapply(tuple$subreferences,
+                                     function (x) x$alias %||% "",
+                                     character(1))
+
+        if (is.null(names(tuple$subreferences))) {
+            # if there are no names for the subvariable elements, fake urls from
+            # the aliases
+            names(tuple$subreferences) <- tuple$subvariables
+        }
+
+        # add a trailling slash to match how urls will look.
```

Comment:
Why is this necessary?

Reply:
The subvariables from a variable entity end with a `/` since they are full URLS (https://github.com/Crunch-io/rcrunch/blob/master/inst/app.crunch.io/api/datasets/1/variables/mymrset.json#L20-L24) 

With cube responses, here, we are faking the url from just the variable id, which lacks the trailing `/` (https://github.com/Crunch-io/rcrunch/blob/master/tests/testthat/cubes/mr-by-mr-different-mrs.json#L231-L234).

This allows the subvariable method for variable tuples to be more general and not special case cubes (at this point specifically: https://github.com/Crunch-io/rcrunch/pull/242/files#diff-d2a62bc0f07894f773e0ab8ecd51becaR57)